### ISSUE TITLE
I think there is a bug in your document

### ISSUE THREAD

**Issue by HaloWang** [bhagat-techind/flutter_sharing_intent#9]
in step 1:

```xml
<string>ShareMedia-$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```

but in step 2:

```swift
let url = URL(string: "SharingMedia-\(hostAppBundleIdentifier)://dataUrl=\(sharedKey)#\(type)")
```

`ShareMedia` vs `SharingMedia`

This problem will kill many people in the chain of sharing implementation.

**Comment by bhagat-techind** [comment]
Oh! My bad. Thanks @HaloWang for helping me to improve this plugin. I will update this tomorrow.

**Comment by HaloWang** [comment]
> Which name is the correct one
> 

make them same. both sharing or both share

**Comment by akilanfidelis** [comment]
Thank you its working now. 
Also one more thing i had to change to make it working was 

Add this which is not mentioned in docs
```
<key>AppGroupId</key>
<string>group.actualgroupid</string>
```

Did you have this in you app ?

**Comment by HaloWang** [comment]
> Thank you its working now. 
> Also one more thing i had to change to make it working was 
> 
> Add this which is not mentioned in docs
> ```
> <key>AppGroupId</key>
> <string>group.actualgroupid</string>
> ```
> 
> Did you have this in you app ?
> 
> 

please check my next issue on this repo, i think the document is miss something about groupid.

**Comment by akilanfidelis** [comment]
Got it. I just noticed, this is only working on a simulator. In real device it doesn't work for me

**Comment by HaloWang** [comment]
> Got it. I just noticed, this is only working on a simulator. In real device it doesn't work for me @bhagat-techind @HaloWang did you face the same

@akilanfidelis 

No, I find my application icon in iOS system share panel on real device.

I have done serveal things:

1. Run share extension by Xcode (attach to Books.app, Safari.app).
2. Run app by xcode.
3. Run app by xcode (release mode).
4. Edit system share panel, move my app icon to the top of the share list.

I'm not share which one is the key. Maybe `flutter run` doesn't install our extension.app to device?

**Comment by akilanfidelis** [comment]
Thanks for the reply back, my issue is mostly sorted. I used browserstack to test on IOS apps, and it didn't work in this only.
It didn't have any issue when I checked with real devices in testflight. So i guess it must be problem with that device only.

**Comment by HaloWang** [comment]
Hi @akilanfidelis ,  have you already integrate this plugin on your android application. I find it is not working as my expected. This plugin add another `MainActivity`(which extends `FlutterActivity`) in its code. So it will start another flutter engine when intent received on android.

What I expect is we can declare an entry-point in our dart code to launch this plugin. And then , this plugin call up / invoke my activity rather than itself. I have some codes in my own `MainActivity`

**Comment by akilanfidelis** [comment]
I have added android first, It worked fine for me without any setup needed. I only had to add listeners in flutter side.

**Comment by HaloWang** [comment]
Well, I find it worked fine for me when there are no native code (however, if you are using provider/riverpod ... the state of your app will fully lost).  But when user share some file to my app and the app is based on some code written in `MainActivity`... 🤯 a catastrophic situation comes.

**Comment by akilanfidelis** [comment]
https://github.com/bhagat-techind/flutter_sharing_intent/issues/9#issuecomment-1467053845

@HaloWang I have this issue similar to this now, I have add factories for nativeAds in android mainactivity class but when app is relaunched through share intent the factory from my mainactivity is not registered but its taking mainactivity of the plugin.
I suddenly remembered the behaviour you said.

**Comment by akilanfidelis** [comment]
I have created a pull request with the fix, i fixed this issue in my app.
https://github.com/bhagat-techind/flutter_sharing_intent/pull/22

**Comment by HaloWang** [comment]
@akilanfidelis 

Basically I think the author of this repo have no time to work on it. Maybe we should maintain this repo by ourself (I also delete Activity class in the codes)

I find flutter team add a document about [iOS App extentsion](https://docs.flutter.dev/platform-integration/ios/app-extensions) which is really like iOS architecture in this repo. But I can not find equal function about Android

**Comment by akilanfidelis** [comment]
Can app extension be used for sharing data between 2 apps ?

**Comment by HaloWang** [comment]
> Can app extension be used for sharing data between 2 apps ?

Not tried but it should be able to do that. app extension and this repo are based on the the same technology on iOS which is call app group

**Comment by didiabel** [comment]
in SharingViewController.swift change
line "SharingMedia-\(hostAppBundleIdentifier)://dataUrl=\(sharedKey)#\(type)"
to
"ShareMedia-\(hostAppBundleIdentifier)://dataUrl=\(sharedKey)#\(type)"

this worked, opened the app for me.

**Comment by bhagat-techind** [comment]
I have fixed this issue in version [1.1.0](https://pub.dev/packages/flutter_sharing_intent/versions/1.1.0)
